### DISCUSSION TITLE
Using Replit as an IDE for this Curriculum

### DISCUSSION THREAD

**Discussion by RandyCHS** [TEALSK12/2nd-semester-introduction-to-computer-science#310] [Q&A]
I realize that since [Replit](https://replit.com/) has moved from Classrooms to [Teams for Education](https://replit.com/teams-for-education) and a new pricing model; TEALS has adapted this curriculum to using [VS Code](https://code.visualstudio.com/) with the [Python extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-python.python).  I'm definitely working to adapt my lessons to VSCode, but I've also invested a lot to work into creating lessons and assignments on Replit.  I also have a K12 Teams for Education license, so I'll continue to develop my curriculum there.  I'm wondering if anyone else will still be using Replit this year and would like to collaborate on and/or share curriculum they have or will create on Replit? 

Randy Macdonald

**Comment by Dan-TEALS** [comment]
The curriculum itself is truly editor/IDE-agnostic. We suggest VS Code as a lightweight, free, open source code editor that is being widely used in industry, but recognize it doesn't work for Chromebooks or managed school devices where local software can't be installed. There really shouldn't be a lot of tweaks required to use replit and I know several of my schools in Arizona plan to continue using it.

I would point out that this curriculum is going into maintenance mode this year, so there won't be any new content created, but we do still plan to fix any issues (and I know there are still some that need to be corrected). Please submit issues in GitHub for anything that needs to be fixed or to make any adjustments you find necessary for using an online IDE, like replit.

**Comment by jdonwells** [comment]
The school I am at is using Chromebooks so I immediately got out my credit card and bought a replit license. The best alternative is pythoneverywhere. It doesn't have all the good stuff replit does.

**Reply by Dan-TEALS** [reply]
I would hope your school is going to reimburse you. Not a huge chunk of change, I agree, but more than I like to see a teacher take out-of-pocket.

**Reply by jdonwells** [reply]
No, the classroom teacher says there is no money. I am willing to pay some money to avoid being ineffective.

**Reply by Dan-TEALS** [reply]
Wow. Thank you, Don. Your classroom teacher is lucky to have you!

**Reply by tmorris1234** [reply]
We are using Code HS to good effect.  It is also free, and it even has some "mini" lessons in the free python course that comes with it, that I have used to great effect.  It's easy to give student's assignments on CodeHS, and view their code live.  Does Replit have these capabilities?  How much money did it cost?

**Reply by jdonwells** [reply]
Unless you pay some money all repls are public. So you can always check on students with or without education level. An education account is $35 per month. For that money you get auto grading,  collaboration where you can add comments to student work and being able to invite based on Google Classroom. 

So far I have not used the auto grading. Right now I will be looking at alternatives for next year because with all repls being public the kids can find my answers or each other's.

**Reply by jdonwells** [reply]
replit for education is free again.

**Reply by Dan-TEALS** [reply]
Thanks for the update, Don.

We're still planning to remove all recommendations for web-based IDEs from the curriculum but it's good to know that replit will be free for all verified EDU institutions!

**Reply by jdonwells** [reply]
repl.it teams for education has been deprecated. Any existing teams still exist. New teams cannot be created. So strip out last year's students and add this year's?

**Comment by jdonwells** [comment]
Another big change at repl.it. It looks like the REPL for Python is oddly difficult to get into. It used to be if your program ran to the end you would get a REPL.  

Now you must make the last line a call to breakpoint(). You get a Python Debugger prompt of (Pdb) . That is essentially a REPL.
